x-pack/plugins/infra/public/components/range_date_picker/index.tsx

```diff
@@ -67,350 +200,370 @@ interface RangeDatePickerState {
+      const { startDate, endDate, recentlyUsed } = this.managedStartEndDateFromType(type, from, to);
+      this.setState(
+        {
+          ...this.state,
+          isPopoverOpen: false,
+          startDate,
+          endDate,
+          recentlyUsed,
+        },
+        () => {
+          if (type) {
+            this.props.onChangeRangeTime(startDate, endDate, true);
+          }
+        }
+      );
+    };
+    private managedStartEndDateFromType(type: string, from?: string, to?: string) {
+      const { intl } = this.props;
+      let { startDate, endDate } = this.state;
+      let recentlyUsed: RecentlyUsed[] = this.state.recentlyUsed;
+      let textJustUsed = type;
+      if (type === 'quick-select') {
+        textJustUsed = intl.formatMessage(
+          {
+            id: 'xpack.infra.rangeDatePicker.lastQuickSelectTimeText',
+            defaultMessage: 'Last {quickSelectTime} {quickSelectUnit}',
+          },
+          {
+            quickSelectTime: this.state.quickSelectTime,
+            quickSelectUnit:
+              this.state.quickSelectTime === 1
+                ? find(singleLastOptions, { value: this.state.quickSelectUnit })!.text
+                : find(pluralLastOptions, { value: this.state.quickSelectUnit })!.text,
+          }
+        );
+        startDate = moment().subtract(this.state.quickSelectTime, this.state
+          .quickSelectUnit as moment.unitOfTime.DurationConstructor);
+        endDate = moment();
+      } else if (type === 'today') {
+        startDate = moment().startOf('day');
+        endDate = moment()
+          .startOf('day')
+          .add(24, 'hour');
+      } else if (type === 'yesterday') {
+        startDate = moment()
+          .subtract(1, 'day')
+          .startOf('day');
+        endDate = moment()
+          .subtract(1, 'day')
+          .startOf('day')
+          .add(24, 'hour');
+      } else if (type === 'this_week') {
+        startDate = moment().startOf('week');
+        endDate = moment()
+          .startOf('week')
+          .add(1, 'week');
+      } else if (type === 'week_to_date') {
+        startDate = moment().subtract(1, 'week');
+        endDate = moment();
+      } else if (type === 'this_month') {
+        startDate = moment().startOf('month');
+        endDate = moment()
+          .startOf('month')
+          .add(1, 'month');
+      } else if (type === 'month_to_date') {
+        startDate = moment().subtract(1, 'month');
+        endDate = moment();
+      } else if (type === 'this_year') {
+        startDate = moment().startOf('year');
+        endDate = moment()
+          .startOf('year')
+          .add(1, 'year');
+      } else if (type === 'year_to_date') {
+        startDate = moment().subtract(1, 'year');
+        endDate = moment();
+      } else if (type === 'date-range' && to && from) {
+        startDate = moment(from);
+        endDate = moment(to);
+      }
+      textJustUsed =
+        textJustUsed && !textJustUsed.includes('Last')
```

Comment:
This doesn't work with custom or pseudo locale. 
I'd move this login at the begging of this function in line 352:
```js
let textJustUsed =
        type === 'date-range' || !type ? type : get(find(commonDates, { id: type }), 'label');
```
also I'd add safe reading of `label` variable using lodash `get` function.